app/controlplane/pkg/biz/casbackend.go

```diff
@@ -336,26 +336,56 @@ func (uc *CASBackendUseCase) Update(ctx context.Context, orgID, id, description
 	}
 
 	var secretName string
+	credentialsUpdated := false
 	// We want to rotate credentials
 	if creds != nil {
 		secretName, err = uc.credsRW.SaveCredentials(ctx, orgID, creds)
 		if err != nil {
 			return nil, fmt.Errorf("storing the credentials: %w", err)
 		}
+		credentialsUpdated = true
 	}
 
-	after, err := uc.repo.Update(ctx, &CASBackendUpdateOpts{
+	// Update the backend without modifying validation status directly
+	// The validation status will be updated through PerformValidation if needed
+	// Don't set validation status here - let PerformValidation handle it
+	updateOpts := &CASBackendUpdateOpts{
 		ID: uuid,
 		CASBackendOpts: &CASBackendOpts{
-			SecretName: secretName, Default: defaultB, Description: description, OrgID: orgUUID,
-			ValidationStatus: CASBackendValidationOK,
-			ValidationError:  ToPtr(""),
+			SecretName:  secretName,
+			Default:     defaultB,
+			Description: description,
+			OrgID:       orgUUID,
 		},
-	})
+	}
+
+	// If we're not updating credentials, preserve the current validation status
+	if !credentialsUpdated {
+		updateOpts.ValidationStatus = before.ValidationStatus
+		updateOpts.ValidationError = before.ValidationError
+	}
+
+	after, err := uc.repo.Update(ctx, updateOpts)
 	if err != nil {
 		return nil, err
 	}
 
+	// If credentials were updated, perform validation to check if they work
+	// This will properly update validation status and send events
+	if credentialsUpdated {
+		if err := uc.PerformValidation(ctx, id); err != nil {
```

Comment:
This part was missing, if we rotate the credentials from the Update, we need to perform a validation to update and send the proper status. The `PerformValidation` takes care of updating the status of the CAS Backend.